packages/datastore/src/sync/index.ts

```diff
@@ -721,10 +721,15 @@ export class SyncEngine {
 							});
 						});
 
+						// null is cast to 0 resulting in unexpected behavior
+						// If newestFullSyncStartedAt is null this is the first sync.
+						// Assume newestStartedAt is is also newest full sync.
 						const msNextFullSync =
-							newestFullSyncStartedAt! +
-							theInterval! -
-							(newestStartedAt! + duration!);
+							newestFullSyncStartedAt! === null
+								? theInterval! - duration!
+								: newestFullSyncStartedAt! +
+								  theInterval! -
+								  (newestStartedAt! + duration!);
```

Comment:
Non-blocking suggestion: the change makes sense, but I think this ternary is getting a bit unruly. 

Might be out of scope here, but the naming of those variables is pretty confusing as well IMO. I would change `theInterval` -> `syncInterval`
`duration` -> `syncDuration`
`newestFullSyncStartedAt` -> `lastFullSyncStartedAt`
`newestStartedAt` -> `lastStartedAt`

Then break apart the ternary like:

```suggestion
						let msNextFullSync;

						if (lastFullSyncStartedAt! === null) {
							msNextFullSync = syncInterval! - syncDuration!;
						} else {
							msNextFullSync =
								lastFullSyncStartedAt! +
								syncInterval! -
								(lastStartedAt! + syncDuration!);
						}
```